sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs

```diff
@@ -109,8 +110,8 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
         internal static IWebJobsBuilder AddServiceBusScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
         {
             builder.Services.AddSingleton(serviceProvider => new ServiceBusScalerProvider(serviceProvider, triggerMetadata));
-            builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider => serviceProvider.GetRequiredService<ServiceBusScalerProvider>());
-            builder.Services.AddSingleton<ITargetScalerProvider>(serviceProvider => serviceProvider.GetRequiredService<ServiceBusScalerProvider>());
+            builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider => serviceProvider.GetServices<ServiceBusScalerProvider>().Where(x => x.GetMonitor().Descriptor.FunctionId == triggerMetadata.FunctionName).First());
```

Comment:
Why do we need to add the two interface singletons? I don't see us injecting this interface anywhere and we've already added the ServiceBusScalerProvider on L112.

Reply:
`ServiceBusScalerProvider` implements both `IScaleMonitorProvider` and `ITargetScalerProvider` and we want to create one instance of `ServiceBusScalerProvider`. Web Jobs SDK needs to resolve both `IScaleMonitorProvider` and `ITargetScalerProvider` so we are adding mapping to the `ServiceBusScalerProvider` instance added.

Reply:
Thanks for that context. Can you explain why we would have more than a single instance of ServiceBusScalerProvider? Isn't there only a single instance of the SB extension in the running process?

Reply:
We are adding ServiceBusScalerProvider for each trigger (if the app has multiple SB triggers).